torchtitan/utils.py

```diff
@@ -34,12 +34,12 @@ def get_device_info():
 device_type, device_module = get_device_info()
 
 
-def dist_max(x: Union[int, float], mesh: DeviceMesh) -> float:
+def dist_max(x: Union[int, float, torch.Tensor], mesh: DeviceMesh) -> float:
     tensor = torch.tensor(x).to(device_type)
```

Comment:
AFAIK these will result in no-ops if a tensor of with the same device type is passed.